py-polars/polars/dataframe/frame.py

```diff
@@ -3959,12 +3963,58 @@ def filter(
         │ 3   ┆ 8   ┆ c   │
         └─────┴─────┴─────┘
 
+        Provided multiple filters using alternative syntax:
+
+        >>> df.filter(pl.col("foo") == 1, pl.col("ham") == "a")
+        shape: (2, 3)
+        ┌─────┬─────┬─────┐
+        │ foo ┆ bar ┆ ham │
+        │ --- ┆ --- ┆ --- │
+        │ i64 ┆ i64 ┆ str │
+        ╞═════╪═════╪═════╡
+        │ 1   ┆ 6   ┆ a   │
+        └─────┴─────┴─────┘
+
+        Use column-supplied filters:
+        >>> df.filter(foo=1, ham="a")
+        shape: (2, 3)
+        ┌─────┬─────┬─────┐
+        │ foo ┆ bar ┆ ham │
+        │ --- ┆ --- ┆ --- │
+        │ i64 ┆ i64 ┆ str │
+        ╞═════╪═════╪═════╡
+        │ 1   ┆ 6   ┆ a   │
+        └─────┴─────┴─────┘
+
         """
-        if _check_for_numpy(predicate) and isinstance(predicate, np.ndarray):
-            predicate = pl.Series(predicate)
+        has_predicates = len(predicates) > 0
```

Comment:
FYI here, Ruff doesn't like `not not predicates`, but polars doesn't like `pl.Expr and x` for existence checks. Not sure if there's a slicker way to do this. Probably doesn't matter.

Reply:
(Rewrote the logic a bit; avoids this issue)